test/integration_test.go

```diff
@@ -745,6 +746,42 @@ digraph {
 	`)))
 }
 
+func TestTerragruntLayerHasFileRunAll(t *testing.T) {
+	t.Parallel()
+
+	tmpEnvPath := copyEnvironment(t, TEST_FIXTURE_LAYER_HAS_FILE)
+
+	environmentPath := fmt.Sprintf("%s/%s/%s", tmpEnvPath, TEST_FIXTURE_LAYER_HAS_FILE, "run-all")
+	layerhasfile := "file.txt"
+
+	var (
+		stdout bytes.Buffer
+		stderr bytes.Buffer
+	)
+	runTerragruntRedirectOutput(t, fmt.Sprintf("terragrunt graph-dependencies --terragrunt-working-dir %s --terragrunt-layer-has-file %s --terragrunt-ignore-external-dependencies", environmentPath, layerhasfile), &stdout, &stderr)
+	output := stdout.String()
+
+	assert.True(t, strings.Contains(strings.TrimSpace(output), strings.TrimSpace(`
+digraph {
+	"app1" ;
+	"app1" -> "module1";
+}
+`)))
+}
+
+func TestTerragruntLayerHasFileWithNotExistingFile(t *testing.T) {
+	t.Parallel()
+
+	tmpEnvPath := copyEnvironment(t, TEST_FIXTURE_LAYER_HAS_FILE)
+
+	environmentPath := fmt.Sprintf("%s/%s/%s", tmpEnvPath, TEST_FIXTURE_LAYER_HAS_FILE, "run-all")
+	layerhasfile := "not-existing-file"
+
+	_, _, err := runTerragruntCommandWithOutput(t, fmt.Sprintf("terragrunt graph-dependencies --terragrunt-working-dir %s --terragrunt-layer-has-file %s", environmentPath, layerhasfile))
```

Comment:
Will be helpful to have integration tests for `run-all` cases, to track this feature will continue to work in future

Reply:
Okey i will add it